File: Leetcode/1663/1663.go
func getSmallestString(n int, k int) string {
	s := make([]rune, n)
	for i := 0; i < n; i++ {
		s[i] = 'a'
	}
	index := n-1
	sum := n
	for sum < k {
		if sum + 25 <= k {
			s[index] = 'z'
			index--
			sum += 25
		} else {
			s[index] += rune(k - sum)
			break
		}
	}
	return string(s)
}

File: Leetcode/1663/1663.cpp
class Solution {
public:
    string getSmallestString(int n, int k) {
        string ans(n, 'a');
        int sum = n;
        int i = n - 1;
        while (sum < k) {
            if (sum + 25 <= k) {
                ans[i] = 'z';
                --i;
                sum += 25;
            } else {
                ans[i] += k - sum;
                break;
            }
        }
        return ans;
    }
};